src/battle_ai_util.c

```diff
@@ -274,7 +274,7 @@ bool32 IsBattlerTrapped(u32 battler, bool32 checkSwitch)
         return FALSE;
     if (checkSwitch && holdEffect == HOLD_EFFECT_SHED_SHELL)
         return FALSE;
-    else if (!checkSwitch && GetBattlerAbility(battler) == ABILITY_RUN_AWAY)
+    else if (!checkSwitch && AI_DATA->abilities[battler] == ABILITY_RUN_AWAY)
```

Comment:
Shouldn't this use the new `AI_GetBattlerAbility` then?

Reply:
the result of `AI_GetBattlerAbility` is stored in `AI_DATA->abilities` so that was a leftover.